rosetta/transactions/compile_transaction.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package transactions
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go-sdk"
+
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+// CompileTransaction creates a complete Flow transaction from the given intent and metadata.
+func (p *Parser) CompileTransaction(intent *Intent, metadata object.Metadata) (*flow.Transaction, error) {
+
+	// Run validation on the block ID. This also fills in missing information.
+	completed, err := p.validate.Block(metadata.ReferenceBlockID)
```

Comment:
I plan on rebasing the feature branch after this to sync it with master. This validation is now a bit different.

Reply:
Quite different, yes.